### PR TITLE
Profile page

### PR DESCRIPTION
**Pull request by Oransj**
There is a TODO which will be there until a meeting at a later time to figure our a good solution. It is there as a reminder for the future.

### CODE REVIEW

File: internal/models/users/users.go [hunk 1]
```diff
@@ -191,3 +191,12 @@ func AssignOptInRankingToUser(db *sql.DB, userID uuid.UUID, optInRanking bool) e
 	)
 	return err
 }
+
+func DeleteUserByUUID(db *sql.DB, userID uuid.UUID) error {
```
Comment (z-rayc) [inline internal/models/users/users.go:195]: By ID; UUID is the data type and should be irrelevant for the function name?

File: internal/web_server/web/views/pages/quiz_pages/user_profile.templ [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        }); 
+    }
+    })
+}
+
+templ UserProfile(u *users.User) {
+	@layout_components.BaseLayout("Nyhetsjeger - Profil") {
+		@showDeleteConfirmationModal()
+		<div class="h-screen flex justify-center items-center">
+			<form class="w-1/3 h-3/4 bg-purple-400 rounded-lg flex flex-col items-center" action="" method="post">
```
Comment (z-rayc) [inline internal/web_server/web/views/pages/quiz_pages/user_profile.templ:36]: No point in having action and post when using HTMX?

File: internal/web_server/web/views/pages/quiz_pages/user_profile.templ [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+					@icons.Dice(16, "currentColor", 28, 28)
+				</button>
+				<div class="flex items-center">
+					<input id="comp-check" class="inline w-6 h-6" type="checkbox" name="competition"/>
+					<p for="comp-check" class="inline ml-2">Bli med i konkurranser</p>
+				</div>
+				<input class="mb-4 mt-5 py-2 px-8 bg-white border-2 rounded-lg font-bold" type="submit" value="Lagre"/>
+				<button hx-post="/auth/logout" class="mb-4 mt-5 py-1 px-8 bg-white border-2 rounded-lg font-bold" type="button">Logg ut</button>
+				<a class="mt-5" href="#">Terms of services</a>
+				<button id="delete-button" hx-delete="" hx-confirm="Sikker på at du ønsker å slette brukeren?" class="text-red-600 font-bold mt-auto mb-4" type="button">SLETT KONTO</button>
```
Comment (z-rayc) [inline internal/web_server/web/views/pages/quiz_pages/user_profile.templ:58]: Why is there an empty hx-delete? Also, duplicate confirmation text, but it is not the same text. `"Sikker på at du ønsker å slette brukeren?"` and `Er du sikker på at du ønsker å slette din konto? Dette kan ikke angres!` If this is unintentional, consider making a variable that is used both places.

**Review by z-rayc — CHANGES_REQUESTED** [review]

File: internal/models/users/users.go (hunk 1 above)
Comment (Molnes) [reply internal/models/users/users.go:195]: UUID is also the actual name of the id we are using, so it doesnt really matter which one we use, as long as it's consistent all over the codebase

Comment (z-rayc) [reply internal/models/users/users.go:195]: Everywhere else we use ID

Comment (Oransj) [reply internal/models/users/users.go]: done

File: internal/web_server/web/views/pages/quiz_pages/user_profile.templ (hunk 2 above)
Comment (Oransj) [reply internal/web_server/web/views/pages/quiz_pages/user_profile.templ:36]: This is a form. I do believe it requires method either way. Empty actions means current path. I do not know how else you can send form data with HTMX.

Comment (z-rayc) [reply internal/web_server/web/views/pages/quiz_pages/user_profile.templ:36]: Everywhere you use `hx-post=""`, `hx-patch=""`, `hx-delete=""` you are already overriding the form action and method. But I suppose there is no harm in letting it stay, just a bit redundant.

**Review by z-rayc — APPROVED** [review]

**Pull request merged**